osbs/build/build_requestv2.py

```diff
@@ -0,0 +1,184 @@
+"""
+Copyright (c) 2018 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import print_function, absolute_import, unicode_literals
+
+import logging
+
+from osbs.build.build_request import BuildRequest
+from osbs.build.user_params import BuildUserParams
+from osbs.exceptions import OsbsValidationException
+from osbs.utils import git_repo_humanish_part_from_uri
+
+logger = logging.getLogger(__name__)
+
+
+class BuildRequestV2(BuildRequest):
+    """
+    Wraps logic for creating build inputs
+    """
+
+    def __init__(self, build_json_store, customize_conf=None):
+        """
+        :param build_json_store: str, path to directory with JSON build files
+        :param customize_conf: str, path to customize configuration JSON
+        """
+        super(BuildRequestV2, self).__init__(build_json_store=build_json_store,
+                                             customize_conf=customize_conf)
+        self.spec = None
+        self.user_params = BuildUserParams(build_json_store)
+        self.osbs_api = None
+
+    # Override
+    def set_params(self, **kwargs):
+        """
+        set parameters in the user parameters
+
+        these parameters are accepted:
+        :param git_uri: str, uri of the git repository for the source
+        :param git_ref: str, commit ID of the branch to be pulled
+        :param git_branch: str, branch name of the branch to be pulled
+        :param base_image: str, name of the parent image
+        :param name_label: str, label of the parent image
+        :param user: str, name of the user requesting the build
+        :param component: str, name of the component
+        :param release: str,
+        :param build_image: str,
+        :param build_imagestream: str,
+        :param build_from: str,
+        :param build_type: str, orchestrator or worker
+        :param platforms: list of str, platforms to build on
+        :param platform: str, platform
+        :param koji_target: str, koji tag with packages used to build the image
+        :param koji_task_id: str, koji ID
+        :param koji_parent_build: str,
+        :param flatpak: if we should build a Flatpak OCI Image
+        :param flatpak_base_image: str, name of the Flatpack OCI Image
+        :param reactor_config_map: str, name of the config map containing the reactor environment
+        :param yum_repourls: list of str, uris of the yum repos to pull from
+        :param signing_intent: bool, True to sign the resulting image
+        :param compose_ids: list of str,
```

Comment:
:param compose_ids, list of int, ODCS composes to use instead of generating them

Reply:
Fixed, thanks.